### ISSUE TITLE
Convert to ESM

### ISSUE THREAD

**Issue by sarahsporck** [digitalfabrik/integreat-app#2404]
**Is your feature request related to a problem? Please describe.**
Some dependencies nowadays migrated their packages to ES modules (ESM). It is trivial to use commonjs modules in ESM, but not the other way around. For us this means we cannot use those dependencies anymore. 
Following [this blogpost](https://molily.de/ecmascript-modules/) most browsers should support ESM modules, with Edge being the last one in 2020. We should at least discuss whether we also want to adapt to ESM or whether we want to wait and see how this develops. 
ESM supports tree-shaking, lazy imports and is compatible with webpack. Regarding react-native compatibility/Mobile compatibility I don't have any information.

**Describe the solution you'd like**
Discuss whether we should/want to migrate to ESM. Most of the work be done automatically

**Describe alternatives you've considered**

**Additional context**
Short summary of the main differences: https://dev.to/abbeyperini/tldr-commonjs-vs-esm-47dk
Longer summary: https://www.hoeser.dev/blog/2023-02-21-cjs-vs-esm/

**Comment by sarahsporck** [comment]
For now start by implementing this for the tools repo. After the PR is open we then evaluate the complexity of this task and whether to convert other repos, too.

**Comment by sarahsporck** [comment]
Progress:

- [ ] tools
- [ ] web
- [ ] native
- [ ] api-client
- [ ] build-configs
- [ ] translations

**Comment by sarahsporck** [comment]
Web to ESM is currently blocked by styled-components update. As well probably the native migration. And as web and native *must* happen before api-client, build-configs and translations (to not break importing files) the whole ticket is on hold.

**Comment by LeandraH** [comment]
We're currently on 6.1.8 of styled-components, so we can probably proceed.

**Comment by LeandraH** [comment]
@lunars97 It's probably a good idea to wait with this ticket until after I'm done with the dependency updates that I'm currently working on

**Comment by steffenkleinle** [comment]
Libraries that migrated to ESM and cannot be upgraded anymore without migrating to ESM:
- [uuid](https://github.com/uuidjs/uuid) >= [12.0.0](https://github.com/uuidjs/uuid/blob/main/CHANGELOG.md)
- [flat](https://github.com/hughsk/flat/) >= [6.0.0](https://github.com/hughsk/flat/releases/tag/v6.0.0)
- [decamelize](https://github.com/sindresorhus/decamelize) >= [6.0.0](https://github.com/sindresorhus/decamelize/releases/tag/v6.0.0)

Indirect dependencies:
- [string-width](https://github.com/sindresorhus/string-width) >= [5.0.0](https://github.com/sindresorhus/string-width/releases/tag/v5.0.0)

**Comment by lunars97** [comment]
Right now jest's status is the following and it hasn't changed since last year https://jestjs.io/docs/ecmascript-modules : 

> Jest ships with experimental support for ECMAScript Modules (ESM). The implementation may have bugs and lack features. 

should we wait till it becomes stable or should we anyway start with it? 🤔

**Comment by steffenkleinle** [comment]
I think we could just attempt this anyway, I guess we'd run into problems pretty soon if any.

**Comment by lunars97** [comment]
> I think we could just attempt this anyway, I guess we'd run into problems pretty soon if any.

sounds reasonable 👍 Should we move it then to next-up?

**Comment by andrew8er** [comment]
I've wasted quiet some time getting jest to work with our vite/esm setup in entitlementcard. I can say that the method outlined in https://jestjs.io/docs/ecmascript-modules is almost certainly the wrong one. What worked was a careful configuration of ts-jest and module related settings in `tsconfig.json`.

I encourage anyone tackling this to have a look the entitlementcard repo.

**Comment by lunars97** [comment]
> I've wasted quiet some time getting jest to work with our vite/esm setup in entitlementcard. I can say that the method outlined in https://jestjs.io/docs/ecmascript-modules is almost certainly the wrong one. What worked was a careful configuration of ts-jest and module related settings in `tsconfig.json`.
> 
> I encourage anyone tackling this to have a look the entitlementcard repo.

really nice, thank you for a hint. I will have a look 🙏
